raiden/ui/startup.py

```diff
@@ -0,0 +1,363 @@
+        if known_given_network_id and known_node_network_id:
+            click.secho(
+                f"The chosen ethereum network '{ID_TO_NETWORKNAME[given_network_id]}' "
+                f"differs from the ethereum client '{ID_TO_NETWORKNAME[node_network_id]}'. "
+                "Please update your settings.",
+                fg='red',
+            )
+        else:
+            click.secho(
+                f"The chosen ethereum network id '{given_network_id}' differs "
+                f"from the ethereum client '{node_network_id}'. "
+                "Please update your settings.",
+                fg='red',
+            )
+        sys.exit(1)
+
+    config['chain_id'] = given_network_id
+    return given_network_id, known_node_network_id
+
+
+def setup_environment(config: Dict[str, Any], environment_type: Environment) -> Environment:
+    """Sets the config depending on the environment type"""
+    # interpret the provided string argument
+    if environment_type == Environment.PRODUCTION:
+        # Safe configuration: restrictions for mainnet apply and matrix rooms have to be private
+        config['environment_type'] = Environment.PRODUCTION
+        config['transport']['matrix']['private_rooms'] = True
+    else:
+        config['environment_type'] = Environment.DEVELOPMENT
+
+    print(f'Raiden is running in {environment_type.value.lower()} mode')
+    return environment_type
+
+
+def setup_contracts_or_exit(
+        config: Dict[str, Any],
+        network_id: int,
+) -> Tuple[Dict[str, Any], bool]:
+    """Sets the contract deployment data depending on the network id and environment type
+
+    If an invalid combination of network id and environment type is provided, exits
+    the program with an error
+    """
+    environment_type = config['environment_type']
+
+    not_allowed = (  # for now we only disallow mainnet with test configuration
+        network_id == 1 and
+        environment_type == Environment.DEVELOPMENT
+    )
+    if not_allowed:
+        click.secho(
+            f'The chosen network ({ID_TO_NETWORKNAME[network_id]}) is not a testnet, '
+            f'but the "development" environment was selected.\n'
+            f'This is not allowed. Please start again with a safe environment setting '
+            f'(--environment production).',
+            fg='red',
+        )
+        sys.exit(1)
+
+    contracts = dict()
+    contract_addresses_known = False
+    contracts_version = environment_type_to_contracts_version(environment_type)
+
+    config['contracts_path'] = contracts_precompiled_path(contracts_version)
+
+    if network_id in ID_TO_NETWORKNAME and ID_TO_NETWORKNAME[network_id] != 'smoketest':
+        deployment_data = get_contracts_deployment_info(
+            chain_id=network_id,
+            version=contracts_version,
+        )
+
+        contracts = deployment_data['contracts']
+        contract_addresses_known = True
+
+    return contracts, contract_addresses_known
```

Comment:
The variable `contract_addresses_known` is unnecessary, if there is no contract deployment info then `contracts` will be an empty dictionary. If the addresses are known or not can be checked with a simple `bool(contracts)` instead of returning a tuple

Reply:
That is the case now and the code can be simplified. If I recall correctly in the past the interface with the contracts repo was a bit different.